### PR TITLE
Finalizer fix

### PR DESCRIPTION
**Pull request by stesee**
- [/] You've read the [Contributor Guide](https://github.com/apache/lucenenet/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://www.apache.org/foundation/policies/conduct.html).
- [/] You've included unit or integration tests for your change, where applicable.
- [/] You've included inline docs for your change, where applicable.
- [/] There's an open issue for the PR that you are making. If you'd like to propose a change, please [open an issue](https://github.com/apache/lucenenet/issues/new/choose) to discuss the change or find an existing issue.

Summary of the changes (Less than 80 chars)

Fixes apache/lucenenet#950

## Description

Dropped the finalizer in IndexReader

### CODE REVIEW

File: src/Lucene.Net/Lucene.Net.csproj [hunk 1]
```diff
@@ -21,117 +21,118 @@
...
-    <InternalsVisibleTo Include="Lucene.Net.Tests.QueryParser" />
-    <InternalsVisibleTo Include="Lucene.Net.Tests.Cli" /> <!-- For lucene-cli -->
-    <InternalsVisibleTo Include="Lucene.Net.Tests.Replicator" />
-    <InternalsVisibleTo Include="Lucene.Net.Tests.Sandbox" />
-    <InternalsVisibleTo Include="Lucene.Net.Tests.Spatial" />
-    <InternalsVisibleTo Include="Lucene.Net.Tests.Suggest" />
-    <InternalsVisibleTo Include="Lucene.Net.Tests.TestFramework" />
-    <InternalsVisibleTo Include="Lucene.Net.Tests.TestFramework.DependencyInjection" />
-  </ItemGroup>
+	<Import Project="$(SolutionDir).build/nuget.props" />
```
Comment (Shazwazza) [inline src/Lucene.Net/Lucene.Net.csproj:24]: IMO formatting should be done in a separate PR. But that said, we should have correct indentation settings for more file types in .editorconfig and default settings for unspecified files. 2 spaces for XML/JSON type files seems to be the standard.

File: NuGet.config [hunk 2]
```diff
@@ -1,4 +1,4 @@
-<?xml version="1.0" encoding="utf-8"?>
+﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (NightOwl888) [inline NuGet.config:1]: Please revert all changes to this file, as these are not related to the finalizer.

File: src/Lucene.Net/Lucene.Net.csproj [hunk 3]
```diff
@@ -21,117 +21,118 @@
...
+
+	<PropertyGroup>
```
Comment (NightOwl888) [inline src/Lucene.Net/Lucene.Net.csproj:26]: Please revert all changes to this file, as these are not related to the finalizer.

**Review by NightOwl888 — CHANGES_REQUESTED** [review]
Thanks for the PR and your willingness to help make our project better.

There are a few changes we would like you to make before we can accept this PR. Please see the comments inline.

File: src/dotnet/Lucene.Net.Tests.CodeAnalysis/Lucene.Net.Tests.CodeAnalysis.csproj [hunk 4]
```diff
@@ -22,7 +22,7 @@
 <Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFramework>net5.0</TargetFramework>
+    <TargetFramework>net6.0</TargetFramework>
```
[the remark this answers is not in the document]
Comment (NightOwl888) [reply src/dotnet/Lucene.Net.Tests.CodeAnalysis/Lucene.Net.Tests.CodeAnalysis.csproj:25]: We will also need to change this line to `net8.0` so our CI will pick up this change: https://github.com/stesee/lucenenet/blob/FinalizerFix/.build/azure-templates/publish-test-results-for-test-projects.yml#L77

[the remark this answers is not in the document]
Comment (stesee) [reply src/dotnet/Lucene.Net.Tests.CodeAnalysis/Lucene.Net.Tests.CodeAnalysis.csproj:25]: isnt the condition redundant?

[the remark this answers is not in the document]
Comment (stesee) [reply src/dotnet/Lucene.Net.Tests.CodeAnalysis/Lucene.Net.Tests.CodeAnalysis.csproj]: The value for TargetFramework in this project is always "net8.0" -> the condtion will always be true.

[the remark this answers is not in the document]
Comment (NightOwl888) [reply src/dotnet/Lucene.Net.Tests.CodeAnalysis/Lucene.Net.Tests.CodeAnalysis.csproj]: When we publish, we publish all target frameworks at the same time at the solution level. It runs this file for every target framework we specify when running the publish command. That overrides what you have specified in this file. We have to explicitly set all of the other target frameworks to false or it will publish and then attempt to run the tests in CI.

Also, publishing the test results is done with templates. So, it needs to be updated to `net8.0` or the test results will be generated, but not uploaded to the control panel. https://github.com/stesee/lucenenet/blob/FinalizerFix/.build/azure-templates/publish-test-results-for-test-projects.yml#L77

[the remark this answers is not in the document]
Comment (stesee) [reply src/dotnet/Lucene.Net.Tests.CodeAnalysis/Lucene.Net.Tests.CodeAnalysis.csproj]: Seems really complex. If that worked in the past we shouldn't change that anyway.

**Review by NightOwl888 — APPROVED** [review]
